### ISSUE TITLE
Some flow events do not start at the correct slice because slices overlap

### ISSUE THREAD

**Issue by yuhao** [google/trace-viewer#1049]
When we decide the start slice of a flow point, we do:

```
var startSlice = thread.sliceGroup.findSliceAtTs(event.ts / 1000);
```

which calls into
```
tr.b.findIndexInSortedClosedIntervals
```

which assumes that "The array of intervals formed by this mapping must be non-overlapping".

However, there are many cases where toplevel slices do overlap. This leads to wrong binding of the start slice.

@natduca @dj2

**Comment by yuhao** [comment]
If we have:

[      A         ][         B          ][         C            ]

{A, ph:"X", ts:"0", dur:"10"}
{B, ph:"X", ts:"10", dur:"10"}
{F, ph:"s", ts:"10"}
{C, ph:"X", ts:"20", dur:"10"}

The end of A is the same as the start of B, and the flow event F starts at B. F's start slice should be B, but is tied to A due to ```findIndexInSortedClosedIntervals```.


Simply changing to use ```findIndexInSortedIntervals``` will not fix it. Consider the following case:

{A, ph:"X", ts:"0", dur:"10"}
{B, ph:"X", ts:"10", dur:"10"}
{F, ph:"s", ts:"20"}
{C, ph:"X", ts:"20", dur:"10"}

The flow event F starts at the end of B, and F's start slice should be B, but will be tied to C if we simply use ```findIndexInSortedIntervals```.

**Comment by dj2** [comment]
So, in those cases, we need to annotate the flow event so we know the binding point. Things that start/end on boundaries are going to be ambiguous.

**Comment by yuhao** [comment]
Even if we annotate F's binding point for start is B, the fact that A and B as well as B and C overlap make it hard to simply use ```thread.sliceGroup.findSliceAtTs``` to bind.

**Comment by dj2** [comment]
We should be able to do some logic using epsilon values and the timestamps to figure it out. Someone just needs to sit down, figure out some of these edge cases and write up a doc on how we deal with finding the binding points.

**Comment by natduca** [comment]
Did the bounds() function turn out to be workable as the solution here? We use that in the slice group builder too.

**Comment by yuhao** [comment]
This will be closed once #1079 is resolved, i.e., with the new flow event API, this shouldn't be an issue.
